src/v/cloud_storage/partition_manifest.cc

```diff
@@ -468,6 +468,11 @@ uint64_t partition_manifest::cloud_log_size() const {
     return _cloud_log_size_bytes;
 }
 
+void partition_manifest::disable_permanently() {
+    _last_offset = model::offset::max();
+    _last_uploaded_compacted_offset = model::offset::max();
+}
```

Comment:
Enumerating some cases so we're clear on expected behavior:

Manifest downloaded, segments transient download failure (timeout) => start the partition with disabled manifest.
* This is only possible in archive-only topics that are configured to not read from cloud.
* Currently there's no guarantee that all segments failed to download in the same way (some may timeout, some may not), and thus there's no guarantee that every replica will be identical.
* If we haven't already, we should consider removing the segment download timeout. If we do that, then we should have a stronger assurance that every replica becomes disabled in the same. 

Manifest downloaded, segments fatal download failure (segment not found) => start the partition with disabled manifest.
* This is only possible in archive-only topics that are configured to not read from cloud
* Every replica should have identical local state since they all fail to download at the same segment. That said, we don't want to continue uploading, since we prefer to preserve existing segment data[1].

Manifest transient download failure (manifest not found) => partition creation retried in reconciliation loop.

Manifest fatal download failure (manifest not found) => partition created empty and not disabled[2].


[1] is this true? Upon restoring the topic, the partition itself will continue to accept new records, but be unable to upload further, which seems risky to proceed with without warning. All further restores of this topic will truncate in the same way, losing any data uploaded post-restore.

I wonder if it makes sense to keep the manifest's last offset as is, but remove all segments starting from where the failed segment is. That way new segments get uploaded past the end of the existing manifest, and we can adjust the start delta so the next offset corresponds to the correct Kafka delta after the first offset? Still probably not a good idea, because it becomes harder to reason about things like whether we can reupload or adjacent segment merge without colliding with existing segments, but it's an option to consider.

[2] Should this actually be disabled? What differentiates this from a missing segment? It seems odd to disable in the missing segment case but enable with missing partition. Maybe we should consider removing the concept of disabled manifest altogether if we can make the errors more deterministic.

Reply:
amazing